### PR TITLE
Added not reserved keyword. Works for both not in and not like

### PR DESCRIPTION
**Pull request by frahman5**
The basic update I made is to add a "Not" field to the NodeIn and NodeLike types and update their assertion methods accordingly. The Not field acts as a flag telling the program whether to query with a normal In/Like or with a Not In/Not Like. If "Not" is set to true, then the assertion method returns the opposite of what it would return otherwise. 

All the other changes just support the above structural update. I updated lexical to be able to process a "not" reserved keyword. And I updated the parser to be able to correctly read in the "not" and create the appropriate NodeIn and NodeLike objects. I also added some tests to parser_test.go to make sure that was all working. 

I chose this route over the other route of creating entirely new NodeNotIn and NodeNotLike objects because:
- it seems simpler, requiring less overall new code. 
- it feels more appropriate, since the 'not' keyword is not really asking the program to create a whole different kind of where condition, but is just _modifying_ the upcoming not or like expression. And so it seems appropriate to reflect that in the code, using the inclusion of a 'not' keyword to _modify_ existing NodeIn or NodeLike objects, rather than create new negative versions of them. 
- it would extend pretty easily if we were to add new reserved keywords, like between. In this model, if you wanted to support 'not between', you would just add a 'Not' field to the NodeBetween object and update the getWhereConds() functions appropriately. 

### Examples
**A query with no where condition**
```
// ♥ go run gitql.go cmd.go autocomplete.go "select author, message from commits limit 5'"
+---------------------+--------------------------------+
|       author        |            message             |
+---------------------+--------------------------------+
| frahman5            | Added not reserved keyword.    |
|                     | Functions for both not in and  |
|                     | not like                       |
+---------------------+--------------------------------+
| Claudson Oliveira   | Update supported go version on |
|                     | travis CI                      |
+---------------------+--------------------------------+
| Claudson Oliveira   | Release a minor release        |
+---------------------+--------------------------------+
| ia                  | all: gofmt (#65)               |
+---------------------+--------------------------------+
| Arumugam Jeganathan | Add 'like' functionality in    |
|                     | where clause (#64)             |
+---------------------+--------------------------------+
```
**A query with a where condition that excludes several of the authors you'd otherwise see**
```
// ♥ go run gitql.go cmd.go autocomplete.go "select author, message from commits where 'Claudson' not in author AND 'Arumugam' not in author limit 5'"
+---------------------+---------------------------------+
|       author        |             message             |
+---------------------+---------------------------------+
| frahman5            | Added not reserved keyword.     |
|                     | Functions for both not in and   |
|                     | not like                        |
+---------------------+---------------------------------+
| ia                  | all: gofmt (#65)                |
+---------------------+---------------------------------+
| Luiz Fernando Peres | Merge pull request #62 from     |
|                     | jsixface/feature/WhereNotEquals |
+---------------------+---------------------------------+
| Luiz Fernando Peres | Merge pull request #61 from     |
|                     | jsixface/bug/NilStarSelect      |
+---------------------+---------------------------------+
| Luiz Fernando Peres | Merge pull request #56 from     |
|                     | jsixface/bug/NilStarSelect      |
+---------------------+---------------------------------+
```

**Using a not like to exclude any commits where the message begins with Add**
```
// ♥ go run gitql.go cmd.go autocomplete.go "select author, message from commits where message not like '%Add' limit 5"
+---------------------+---------------------------------+
|       author        |             message             |
+---------------------+---------------------------------+
| Claudson Oliveira   | Update supported go version on  |
|                     | travis CI                       |
+---------------------+---------------------------------+
| Claudson Oliveira   | Release a minor release         |
+---------------------+---------------------------------+
| ia                  | all: gofmt (#65)                |
+---------------------+---------------------------------+
| Luiz Fernando Peres | Merge pull request #62 from     |
|                     | jsixface/feature/WhereNotEquals |
+---------------------+---------------------------------+
| Arumugam J          | register walk type              |
+---------------------+---------------------------------+
```

@luizperes :)

### CODE REVIEW

File: lexical/tokens.go [hunk 1]
```diff
@@ -26,6 +26,8 @@ const T_PARENTH_R = 23
 const T_IN = 24
 const T_ASC = 25
 const T_LIKE = 26
+const T_NOT = 27
+const T_NOT_OR_T_LIKE = 28
```
Comment (cloudson) [inline lexical/tokens.go]: Did your Idea actually was `T_IN_OR_T_LIKE`, isn't?  I'm not too comfortable seeing a token that is used just for error messages. I prefer to suggest just one of the options (T_NOT) and maybe we can discuss better in another thread how to improve the parser error to show more then one expected tokens. What do you think?

File: parser/parser_test.go [hunk 2]
```diff
@@ -230,6 +230,154 @@ func TestWhereWithNotEqualCompare(t *testing.T) {
 	}
 }
 
+func TestWhereWithIn(t *testing.T) {
```
Comment (cloudson) [inline parser/parser_test.go]: Wow, thanks a lot for that ❤️

File: parser/parser_test.go [hunk 3]
```diff
@@ -230,6 +230,154 @@ func TestWhereWithNotEqualCompare(t *testing.T) {
...
+	if rValue.Value() != "message" {
+		t.Errorf("RValue should be 'message'")
+	}
+
+
+
+}
+
+func TestWhereWithNotIn(t *testing.T) {
+	New("Select message from commits where 'react' not in message")
```
Comment (cloudson) [inline parser/parser_test.go]: Vue forever (?) 😆

Comment (frahman5) [reply parser/parser_test.go]: lol

File: parser/parser_test.go (hunk 2 above)
Comment (frahman5) [reply parser/parser_test.go]: :)

File: lexical/tokens.go (hunk 1 above)
Comment (frahman5) [reply lexical/tokens.go]: Yeah I wasn't too happy about that either. I didn't want to create an entirely different SyntaxError function though, so I squeezed in that token instead. I think you're right now. I'll change it to just suggest T_NOT for now, and we can leave the issue of leaving better parser errors for another day.

Comment (frahman5) [reply lexical/tokens.go]: Just letting you know I pushed another commit with this fix :)

File: .idea/gitql.iml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (cloudson) [inline .idea/gitql.iml]: Thank you for the change but you commit some IDE files :/ . Can you remove them?

File: lexical/tokens.go [hunk 5]
```diff
@@ -58,6 +59,7 @@ func allocMapTokenNames() {
 			T_IN:               "T_IN",
 			T_EOF:              "T_EOF",
 			T_ASC:              "T_ASC",
+			T_NOT: 				"T_NOT",
```
Comment (luizperes) [inline lexical/tokens.go]: are these extra spaces only on github? otherwise, could you align them?

**Comment by frahman5** [comment]
@luizperes @cloudson requested changes made. I've removed the IDE files and fixed the spacing on tokens. go :)

**Comment by cloudson** [comment]
Thank you again 🤘

**Comment by frahman5** [comment]
👍 😃

**Comment by jsixface** [comment]
I think this would close #49

**Pull request merged**